Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -1049,6 +1141,79 @@ open class ComposerVC: _ViewController,
         }
     }
 
+    // MARK: - VoiceRecordingDelegate
+
+    public func voiceRecording(
+        _ voiceRecordingVC: VoiceRecordingVC,
+        addAttachmentFromLocation location: URL,
+        duration: TimeInterval,
+        waveformData: [Float]
+    ) {
+        var extraData: [String: RawJSON] = [:]
+        extraData.duration = duration
+        extraData.waveform = waveformData
```

Comment:
I see, it is because the duration and waveform come from the app that we are using extra data for it? I'm not sure if there is any limitation here, but ideally, both duration and waveform should be part of the Payload since extra data is meant for customers to use it 🤔 

Keep in mind, that extra data is just kown/unknown properties. There is not a specific field for extra data, basically whatever data we don't know from a payload, it gets parsed to extra data.

Reply:
This is now being handled in the other PR :)